catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -158,7 +234,48 @@ public Object handle(Request request, Response response) throws Exception {
       return ImmutableMap.of("message", "Service not found");
     }
 
-    CqlQueryResponse cqlQueryResponse = cqlQueryUtil.executeCqlQuery(cqlRequest);
+    List<Result> results =
+        cqlRequests
+            .stream()
+            .map(
+                cqlRequest -> {
+                  CqlQueryResponse cqlQueryResponse = null;
+                  try {
+                    cqlQueryResponse = cqlQueryUtil.executeCqlQuery(cqlRequest);
+                  } catch (Exception e) {
+                    LOGGER.debug("Error fetching cql request for {}", cqlRequest.getSrc());
+                    return null;
+                  }
+                  return cqlQueryResponse.getQueryResponse().getResults();
+                })
+            .filter(cqlResults -> CollectionUtils.isNotEmpty(cqlResults))
+            .flatMap(cqlResults -> cqlResults.stream())
+            .collect(Collectors.toList());
+
+    results.sort(getResultComparators(cqlTransformRequest.getSorts()));
+
+    results =
+        results.size() > cqlTransformRequest.getCount()
+            ? results.subList(0, cqlTransformRequest.getCount())
+            : results;
+
+    results =
+        CollectionUtils.isEmpty(cqlTransformRequest.getHiddenResults())
+            ? results
+            : results
+                .stream()
+                .filter(
+                    result ->
+                        !cqlTransformRequest
+                            .getHiddenResults()
+                            .contains(result.getMetacard().getId()))
+                .collect(Collectors.toList());
+
+    QueryResponse combinedResponse =
+        new QueryResponseImpl(
+            new QueryRequestImpl(new QueryImpl(ECQL.toFilter(cqlRequests.get(0).getCql()))),
+            results,
```

Comment:
so this `results` is a subset fo the total results, if the results exceeded the size of the request, right? I guess there's no way to only get the number that we want, i.e. `cqlTransformRequest.getCount()` number of results